xarray/backends/zarr.py

```diff
@@ -412,6 +421,8 @@ def store(
         self.set_variables(
             variables_encoded, check_encoding_set, writer, unlimited_dims=unlimited_dims
         )
+        if self._save_consolidated:
+            zarr.consolidate_metadata(self.ds.store)
```

Comment:
Why did you move this here, rather than where it used to be (in `close()`)?

Reply:
This is only relevant when writing data, not when reading. So it felt more appropriate to me to put it in a dedicated method for writing.

(We don't support calling Backend methods directly to store data, so I don't think there should be backwards compatibility constraints.)